File: apps/problems/Kth-Smallest/boilerplate-full/function.cpp

            #include <stdio.h>
            #include <stdlib.h>
            #include <stdbool.h>
            #include <string.h>
            #include <limits.h>
            
            ##USER_CODE_HERE##
            
            int main() {
                FILE *file = fopen("/dev/problems/Kth-Smallest/tests/inputs/##INPUT_FILE_INDEX##.txt", "r");
                vector<string> lines;
                string line;
                while (fgets(line, sizeof(line), file)) lines.push_back(line);
                fclose(file);
                int size_A;
  fscanf(file, "%d", &size_A);
  int* A[size_A];
  for (int i = 0; i < size_A; i++) fscanf(file, "%d", &A[i]);
  int size_B;
  fscanf(file, "%d", &size_B);
  int* B[size_B];
  for (int i = 0; i < size_B; i++) fscanf(file, "%d", &B[i]);
  int K;
  fscanf(file, "%d", &K);
                int result = solve(A, B, K);
                fprintf(stdout, "%d\n", result);
                return 0;
                }
            

File: apps/problems/Kth-Smallest/boilerplate-full/function.rs
use std::fs::read_to_string;
    use std::io::{self};
    use std::str::Lines;
    
    ##USER_CODE_HERE##
    
    fn main() -> io::Result<()> {
      let input = read_to_string("/dev/problems/Kth-Smallest/tests/inputs/##INPUT_FILE_INDEX##.txt")?;
      let mut lines = input.lines();
      let size_A: usize = lines.next().and_then(|line| line.parse().ok()).unwrap_or(0);
	let A: Vec<i32> = parse_input(lines, size_A);
  let size_B: usize = lines.next().and_then(|line| line.parse().ok()).unwrap_or(0);
	let B: Vec<i32> = parse_input(lines, size_B);
  let K: i32 = lines.next().unwrap().parse().unwrap();
      let result = solve(A, B, K);
      println!("{}", result);
      Ok(())
    }
fn parse_input(mut input: Lines, size_arr: usize) -> Vec<i32> {
        let arr: Vec<i32> = input
            .next()
            .unwrap_or_default()
            .split_whitespace()
            .filter_map(|x| x.parse().ok())
            .collect();
    
        if size_arr == 0 {
            Vec::new()
        } else {
            arr
        }
    }
    